packages/core/src/lib/components/toggle.tsx

```diff
@@ -18,25 +21,48 @@ const Toggle: React.FC<ToggleProps> = ({ children, ...props }) => {
 
   const toggleOpen = useCallback(() => setOpen((prevOpen) => !prevOpen), []);
 
+  let iconElement: React.ReactNode = null;
+  const otherChildren: React.ReactNode[] = [];
+
+  React.Children.forEach(children, (child) => {
+    if (React.isValidElement(child) && child.type === Toggle.Icon) {
+      iconElement = child.props.children;
+    } else {
+      otherChildren.push(child);
+    }
+  });
+
+  if (!iconElement) {
+    iconElement = <div className="notion-toggle-button-arrow" />;
```

Comment:
p3: you just declare default iconElement on 

let iconElement = {{ default }} // instead of assigning `null`